src/toil_scripts/lib/files.py

```diff
@@ -0,0 +1,77 @@
+from contextlib import closing
+import os
+import errno
+import tarfile
+import shutil
+
+
+# FIXME: replace with bd2k.util.files.mkdir_p
+def mkdir_p(path):
+    """
+    It is Easier to Ask for Forgiveness than Permission
+
+    :param str path: path to directory to create
+    """
+    try:
+        os.makedirs(path)
+    except OSError as exc:
+        if exc.errno == errno.EEXIST and os.path.isdir(path):
+            pass
+        else:
+            raise
+
+
+def tarball_files(tar_name, fpaths, work_dir='.', prefix=''):
+    """
+    Creates a tarball from a group of files
+
+    :param str tar_name: Name of tarball
+    :param list[str] fpaths: File paths to include in the tarball
+    :param str work_dir: Current working directory
+    :param str prefix: Optional prefix for files in tarball
+    """
+    with tarfile.open(os.path.join(work_dir, tar_name), 'w:gz') as f_out:
+        for fpath in fpaths:
+            arcname = prefix + os.path.basename(fpath)
+            f_out.add(fpath, arcname=arcname)
+
+
+def move_files(filepaths, output_dir):
+    """
+    Moves files from the working directory to the output directory.
+
+    :param str output_dir: Output directory
+    :param list[str] filepaths: File paths to move
+    """
+    for fpath in filepaths:
+        dest = os.path.join(output_dir, os.path.basename(fpath))
+        shutil.move(fpath, dest)
+
+
+def consolidate_tarballs_job(job, **fname_to_id):
+    """
+    Combine the contents of separate tarballs into one.
+    Subdirs within the tarball will be named the keys in **fname_to_id
+
+    :param dict[str,str] fname_to_id: Key word arguments of the form: file_name_prefix=FileStoreID
+    :return str: FileStoreID
```

Comment:
Does this shorthand actually work? Does PyCharm pick it up? You can find out by hovering over a reference to that function or by writing a invocation and appending `.`. If that pops up a list of the methods in `str`, you should be good. Otherwise, use `:rtype:` separately.